port/unix/omrsysinfo.c

```diff
@@ -4465,70 +4446,108 @@ omrsysinfo_cgroup_subsystem_iterator_hasNext(struct OMRPortLibrary *portLibrary,
 	return check;
 }
 
+int32_t
+omrsysinfo_cgroup_subsystem_iterator_metric_key(struct OMRPortLibrary *portLibrary, const struct OMRCgroupMetricIteratorState *state, const char **metricKey)
+{
+	int32_t rc = OMRPORT_ERROR_SYSINFO_CGROUP_SUBSYSTEM_METRIC_NOT_AVAILABLE;
+#if defined(LINUX) && !defined(OMRZTPF)
+	const struct OMRCgroupSubsystemMetricMap *subsystemMetricMap = NULL;
+	rc = 0;
+	switch (state->subsystemid) {
+	case OMR_CGROUP_SUBSYSTEM_MEMORY:
+		subsystemMetricMap = omrCgroupMemoryMetricMap;
+		break;
+	case OMR_CGROUP_SUBSYSTEM_CPU:
+		subsystemMetricMap = omrCgroupCpuMetricMap;
+		break;
+	case OMR_CGROUP_SUBSYSTEM_CPUSET:
+		subsystemMetricMap = omrCgroupCpusetMetricMap;
+		break;
+	default:
+		rc = OMRPORT_ERROR_SYSINFO_CGROUP_SUBSYSTEM_UNAVAILABLE;
+		goto _end;
+	}
+	if (state->multiLineCounter < subsystemMetricMap[state->count].metricElementsCount){
+		*metricKey = (subsystemMetricMap[state->count].metricInfoElementList + state->multiLineCounter)->metricTag;
```

Comment:
Can the `rc = 0` occur here so that all the "success" case is together?

Reply:
moved `rc = 0`